base/set.jl

```diff
@@ -191,6 +191,62 @@ function unique(f::Callable, C)
 end
 
 """
+    unique!(c)
+Remove duplicate items and return the modified collection, as determined by
+[`isequal`](@ref).
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A);
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+```
+"""
+function unique!(c)
+    if issorted(c) || issorted(reverse(c))
+        idxs = eachindex(c)
+        # When we loop through c, we only need to remember one state and one
+        # item in the collection.
+        currentstate = start(idxs)
+        currentitem = c[currentstate]
```

Comment:
Right, originally I had a function more like this:
```julia
function uniquejd!(itr)  # I used my initials to distinguish the function.
    seen = Set{eltype(itr)}()
    count = 0  # Number of unique values seen so far.
    for x in itr
        @inbounds if ∉(x, seen)
            count += 1            
            push!(seen, x)
            itr[count] = x
        end
    end
    resize!(itr, count)
end
```
Shall I revert to that idea? It is also really easy to deal with the sorted case using the above idea.